lib/lunchbot_web/live/shared_cart.ex

```diff
@@ -0,0 +1,70 @@
+defmodule LunchbotWeb.SharedCart do
+  @moduledoc false
+
+  use LunchbotWeb, :live_view
+
+  import Ecto.Query
+  import Ecto.Changeset
+  alias Lunchbot.LunchbotData.Office
+  alias Lunchbot.LunchbotData.OfficeLunchOrder
+  alias LunchbotWeb.SharedCart.OfficeSelector
+  alias LunchbotWeb.SharedCart.OfficeSelectorInterfaceContext
+
+  def mount(_params, _session, socket) do
+    if connected?(socket), do: Lunchbot.LunchbotData.OrderItem.subscribe()
+    {:ok, fetch(socket) |> assign_office() |> assign_changeset()}
+  end
+
+  def assign_office(socket) do
+    socket
+    |> assign(:office_selector, %OfficeSelector{})
+  end
+
+  def assign_changeset(%{assigns: %{office_selector: office_selector}} = socket) do
+    socket
+    |> assign(:changeset, OfficeSelectorInterfaceContext.change_office(office_selector))
+  end
+
+  # Fetch DB data, assign to socket watched by this LiveView
+  defp fetch(socket) do
+    office_lunch_orders = get_lunch_order_for_office()
+
+    offices = Lunchbot.Repo.all(Office)
```

Comment:
Can we place all the references to the DB and to Repo in `LunchbotData.ex` as separate functions to keep the contexts separate?

Reply:
Yeah I was thinking either that or making a `shared_cart_controller.ex` and pulling some Repo stuff out to that. LunchbotData is an idea but I think since some of the functions using Repo fetches modify the socket I should put them in a controller.